lib/NativeCode/DynamicLibraryLoaderHelper/NativeRender/dllmain.cpp

```diff
@@ -44,161 +47,126 @@ extern "C"
+        if (argument_strings[i]._Starts_with(sandboxArgName))
         {
             match = &sandboxArgName;
         }
-        else if(argStrings[i]._Starts_with(egsArgName))
+        else if (argument_strings[i]._Starts_with(egsArgName))
         {
             match = &egsArgName;
         }
         if (match != nullptr)
         {
-            std::string sandboxArg = argStrings[i].substr(match->length());
+            std::string sandboxArg = argument_strings[i].substr(match->length());
             if (!sandboxArg.empty())
             {
-                pew::eos::logging::log_inform(("Sandbox ID override specified: " + sandboxArg).c_str());
+                logging::log_inform(("Sandbox ID override specified: " + sandboxArg).c_str());
                 eos_config.sandboxID = sandboxArg;
             }
         }
     }
-#endif
 
     //check if a deployment id override exists for sandbox id
     for (unsigned i = 0; i < eos_config.sandboxDeploymentOverrides.size(); ++i)
     {
         if (eos_config.sandboxID == eos_config.sandboxDeploymentOverrides[i].sandboxID)
         {
-            pew::eos::logging::log_inform(("Sandbox Deployment ID override specified: " + eos_config.sandboxDeploymentOverrides[i].deploymentID).c_str());
+            logging::log_inform(("Sandbox Deployment ID override specified: " + eos_config.sandboxDeploymentOverrides[i].deploymentID).c_str());
             eos_config.deploymentID = eos_config.sandboxDeploymentOverrides[i].deploymentID;
         }
     }
 
-#if PLATFORM_WINDOWS
     std::string deploymentArgName = "-eosdeploymentid=";
     std::string egsDeploymentArgName = "-epicdeploymentid=";
-    for (unsigned i = 0; i < argStrings.size(); ++i)
+    for (unsigned i = 0; i < argument_strings.size(); ++i)
     {
         std::string* match = nullptr;
-        if (argStrings[i]._Starts_with(deploymentArgName))
+        if (argument_strings[i]._Starts_with(deploymentArgName))
         {
             match = &deploymentArgName;
         }
-        else if (argStrings[i]._Starts_with(egsDeploymentArgName))
+        else if (argument_strings[i]._Starts_with(egsDeploymentArgName))
         {
             match = &egsDeploymentArgName;
         }
         if (match != nullptr)
         {
-            std::string deploymentArg = argStrings[i].substr(match->length());
+            std::string deploymentArg = argument_strings[i].substr(match->length());
             if (!deploymentArg.empty())
             {
-                pew::eos::logging::log_inform(("Deployment ID override specified: " + deploymentArg).c_str());
+                logging::log_inform(("Deployment ID override specified: " + deploymentArg).c_str());
                 eos_config.deploymentID = deploymentArg;
             }
         }
     }
+}
+
+// Called by unity on load. It kicks off the work to load the DLL for Overlay
+#if PLATFORM_32BITS
+#pragma comment(linker, "/export:UnityPluginLoad=_UnityPluginLoad@4")
```

Comment:
Is the @4 a magic number? Where does that come from? Is that an argument to the linker?

Reply:
I honestly do not know - it's something that was there already (GitHub is interpreting it as an add because of how extensively the file was changed, but the line existed in the old implementation).